pkg/api-server/resource_endpoints.go

```diff
@@ -438,3 +453,103 @@ func (r *resourceEndpoints) readOnlyMessage() string {
 		return k8sReadOnlyMessage
 	}
 }
+
+func (r *resourceEndpoints) matchingDataplanesForPolicy() restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		policyName := request.PathParameter("name")
+		meshName := r.meshFromRequest(request)
+
+		if err := r.resourceAccess.ValidateGet(
+			request.Request.Context(),
+			model.ResourceKey{Mesh: meshName, Name: policyName},
+			r.descriptor,
+			user.FromCtx(request.Request.Context()),
+		); err != nil {
+			rest_errors.HandleError(request.Request.Context(), response, err, "Access Denied")
+			return
+		}
+		page, err := pagination(request)
+		if err != nil {
+			rest_errors.HandleError(request.Request.Context(), response, err, "Could not retrieve policy")
+			return
+		}
+		nameContains := request.QueryParameter("name")
+
+		policyResource := r.descriptor.NewObject()
+		if err := r.resManager.Get(request.Request.Context(), policyResource, store.GetByKey(policyName, meshName)); err != nil {
+			rest_errors.HandleError(request.Request.Context(), response, err, "Could not retrieve policy")
+			return
+		}
+
+		var dependentTypes []model.ResourceType
+		if r.descriptor.IsTargetRefBased {
+			dependentTypes = []model.ResourceType{v1alpha1.MeshHTTPRouteType, mesh.MeshGatewayType}
+		} else if r.descriptor.Name == mesh.MeshGatewayRouteType {
+			dependentTypes = []model.ResourceType{mesh.MeshGatewayType}
+		}
+		dependentResources := xds_context.NewResources()
+		for _, dependentType := range dependentTypes {
+			hl, err := registry.Global().NewList(dependentType)
+			if err != nil {
+				rest_errors.HandleError(request.Request.Context(), response, err, "failed inspect")
+				return
+			}
+			if err := r.resManager.List(request.Request.Context(), hl, store.ListByMesh(meshName)); err != nil {
+				rest_errors.HandleError(request.Request.Context(), response, err, "failed inspect")
+				return
+			}
+			dependentResources.MeshLocalResources[dependentType] = hl
+		}
+		filter := func(rs model.Resource) bool {
```

Comment:
Could we extract this part until the response is prepared to a separate method like `GetAffectedDataplanes`?

Reply:
I think we could but it'd be a function generator with a bunch of params (dependent resources and policy at least)